### DISCUSSION TITLE
Bind PSK with the connection section

### DISCUSSION THREAD

**Discussion by Bearww** [strongswan/strongswan#972] [Q&A]
Hello community,

I'm trying to setup remote sites via multiple swanctl.conf(s) on my router to accept other sites via PSK.
If there are multiple IKE configurations with the same ID but different secrets, only one secret will be matched.
Is there a way to separate the ID/PSK to different connections?
Ex: Two different sites with different cipher suites and IPs but using the same ID to connect my router.

Best Regards!

**Comment by tobiasbrunner** [comment]
Generally, only configure the remote identity for your secrets, not the local identity that might be the same. If the remote identity is the same too for both connections, you might want to reconsider that as it could cause other issues too (e.g. with uniqueness checks).

**Reply by Bearww** [reply]
Thank you for your reply, @tobiasbrunner!

Sorry, I'm a little confused about this:
> you might want to reconsider that as it could cause other issues too (e.g. with uniqueness checks).

If two connections can be distinguished by IKE (e.g. local/remote IP (range or subnet), proposal), can I consider them to be different group of peers as defined in RFC 4301 PAD?

Many appreciations!

**Reply by tobiasbrunner** [reply]
No, the uniqueness check only uses the identities when comparing IKE_SAs.

**Reply by Bearww** [reply]
Do you mean peers assert their IDs via INITIAL_CONTACT?

**Reply by tobiasbrunner** [reply]
That's one aspect (and forces a uniqueness check unless it's completely disabled on the responder) but uniqueness can also be enforced if the notify is not received (see the `<conn>.unique` option in [swanctl.conf](https://docs.strongswan.org/strongswan-docs/5.9/swanctl/swanctlConf.html)).

**Reply by Bearww** [reply]
Curious about this situation:
> If there are multiple IKE configurations with the same ID but different secrets, only one secret will be matched.

`<conn>.unique` doesn't seem to be useful as peers may suffer MAC failures due to wrong PSK.
Of course, I need to reconsider it first.

**Reply by tobiasbrunner** [reply]
What confuses you? Secrets are looked up based on the identities (those of the IKE_SA are matched against those assigned to a secret), a single match is enough. So you can't have different secrets for two connections if the identities are the same.

**Reply by Bearww** [reply]
Yeah. I know that only one identity can exist in PAD at a time.
Since IKE daemon chooses the appropriate IKE connection by ranking using external IPs, proposals, etc., is it possible to have multiple PADs?
Or do you know of any VPN products that support this?
Thank you so much for your patience in replying to these curious things.

**Reply by tobiasbrunner** [reply]
Sorry, I've no idea what exactly you're asking.
